src/Payment/PaymentTests/APITests.cs

```diff
@@ -0,0 +1,105 @@
+using DAL.Payments;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using PaymentAPI.Controllers;
+using PaymentApplication.Commands;
+using PaymentApplication.Queries;
+using TestsCore;
+
+namespace PaymentTests
+{
+	public class APITests
+	{
+		[Fact]
+		public async Task GetById_NullResult_NotFound()
+		{
+			var mediator = new Mock<IMediator>();
+			mediator.Setup(m => m.Send(It.IsAny<GetPaymentQuery>(), It.IsAny<CancellationToken>()))
+				.ReturnsAsync((Payment)null);
+			var controller = GetController(mediator.Object);
+
+			var result = await controller.GetById(123);
+
+			Assert.NotNull(result);
+			Assert.IsType<NotFoundObjectResult>(result);
+		}
+
+		[Fact]
+		public async Task GetById_WithResult_OkWithPayment()
+		{
+			var mediator = new Mock<IMediator>();
+			var paymentId = 123;
+			mediator.Setup(m => m.Send(It.IsAny<GetPaymentQuery>(), It.IsAny<CancellationToken>()))
+				.ReturnsAsync(new Payment { Id = paymentId });
+			var controller = GetController(mediator.Object);
+
+			var result = await controller.GetById(666);
+			var payment = (result as OkObjectResult)?.Value as Payment;
+
+			Assert.NotNull(payment);
+			Assert.Equal(paymentId, payment.Id);
+		}
+
+		[Fact]
+		public async Task Fail_SuccessResult_Ok()
+		{
+			await ProcessPaymentCommandTestOk(false);
+		}
+
+		[Fact]
+		public async Task Fail_FailedResult_BadRequestWithMessage()
+		{
+			await ProcessPaymentCommandTestBadRequest(false);
+		}
```

Comment:
Why not just use testcase parameters?

Reply:
They use different methods and using test case parameters would be confusing, imho. We can't add message to failed assert so which method failed will become a mystery until you go into code.